app/src/main/java/fr/gaulupeau/apps/Poche/ui/ManageArticleTagsActivity.java

```diff
@@ -88,16 +88,13 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         setContentView(R.layout.activity_manage_article_tags);
 
         String[] currentTagsArray = null;
-
-        Intent intent = getIntent();
-        String text = intent.getStringExtra(PARAM_TAG_LABEL);
-        if (
-                text == null
-                && savedInstanceState != null
-            ) {
+        String text;
+        if (savedInstanceState != null) {
             discoveredArticleId = savedInstanceState.getInt(STATE_DISCOVERED_ARTICLE_ID, -1);
             currentTagsArray = savedInstanceState.getStringArray(STATE_CURRENT_TAGS);
             text = savedInstanceState.getString(STATE_CURRENT_TEXT);
+        } else {
+            text = getIntent().getStringExtra(PARAM_TAG_LABEL);
```

Comment:
I'm not exactly sure what `savedInstanceState` is, but this seems to change the precedence. I assumed that  `savedInstanceState` was, maybe, a previously entered tag, but not added, before switching to another app.

In this case, and if my assumption is correct, I figured that it might make more sense to ignore the previously entered tag, and show the new selection instead, which is the purpose of the intent.

Reply:
`savedInstanceState` is only present when the activity is recreated (like when screen orientation changes), it contains saved state which may be newer (you already edited the string) than the original string in the intent.

That may be complicated with non-standard activity parameters (affinities and stuff), but we don't have anything like that at the moment.